streams/src/main/java/org/apache/kafka/streams/processor/internals/StandbyTask.java

```diff
@@ -97,31 +96,27 @@ public void commit() {
     }
 
     @Override
-    public void close() {
-        //no-op
+    public void suspend() {
+        commit();
     }
 
     @Override
-    public void initTopology() {
-        //no-op
+    public void close() {
+        try {
+            commit();
+            closeStateManager(true);
+        } catch (final RuntimeException e) {
```

Comment:
Hmm, this pattern is a bit weird. As @guozhangwang initially said, it's as likely to introduce issues than fix issues. `closeStateManager` should just do the right thing. If we need to change the `catch` blocks, we should do it there.